### ISSUE TITLE
Ember-data module paths have mixed forward and backslashes when building for production on Windows.

### ISSUE THREAD

**Issue by mattmazzola** [ember-cli/ember-cli#5577]
Hi,
I tried both on ember-cli 2.3.0 and 2.4.1 and there seems to be problems with the way the modules paths are defined in the module's exports list.

Steps to reproduce:
1. Create new ember app
`ember new testproject`
2. Move to project folder
`cd testproject`
3. Serve for production
`ember s --environment production`

When you view the app in the browser there is error in the console:
Could not find module `ember-data/-private\system\references\record` imported from `ember-data/-private/system/references`

If I run the app in normal mode everything works fine, but in production I can look at the names of the modules in the vendor file and see that:

There is the record module defined like this:
`define("ember-data/-private/system/references/record",["exports","ember","ember-data/-private\\system\\references\\reference"],function(e,t,r)`

And a module above it referencing it as dependency:
`define("ember-data/-private/system/references",["exports","ember-data/-private\\system\\references\\record"`

And since references modules is trying to depend on the record module which exists but uses a name with different slashes I assume it can't find it.

TL:DR;
`ember-data/-private\\system\\references\\record` does not equal `ember-data/-private/system/references/record`

Also important to note I was running on windows:
```
ember -v
version: 2.3.0
node: 4.2.1
os: win32 x64
```

Someone else noted that if they downgraded ember-data from 2.3.3 to 2.3.0 it would work but I wasn't able to verify.  It seems like somehow the paths for these simulated private modules are resolved using backslashes and all the other paths use forward slashes.

**Comment by KrisOlszewski** [comment]
Duplicate of #5542.

**Comment by stefanpenner** [comment]
thanks @KrisOlszewski
